src/main/scala/tech/cryptonomic/conseil/tezos/ApiOperations.scala

```diff
@@ -849,16 +843,39 @@ object ApiOperations {
     * @return
     */
   def averageFee(filter: Filter): Try[Fees] = Try {
+    val action = for {
+      fee <- Tables.Fees
+      if filterOperationKindsForFees(filter, fee)
+    } yield (fee.low, fee.medium, fee.high, fee.timestamp, fee.kind)
+    val op = dbHandle.run(action.distinct.sortBy(_._4.desc).take(1).result)
+    val results = Await.result(op, Duration.apply(awaitTimeInSeconds, SECONDS)).head
+    Fees(results._1, results._2, results._3, results._4, results._5)
+  }
+
+
+  /**
+    * Given the operation kind, return range of fees and timestamp for that operation.
+    * @param kind
+    * @return
+    */
+  def averageFee(kind: String): Option[Fees] = {
+    // "seed_nonce_revelation", ﻿"delegation", "transaction", "activate_account", "origination", "reveal", "double_endorsement_evidence", "endorsement"
+    val operationKinds = Set[String]{kind}
     val action = for {
       o <- Tables.Operations
-      if filterOperationKinds(filter, o)
+      if o.kind.inSet(operationKinds)
     } yield (o.fee, o.timestamp)
-    val op = dbHandle.run(action.distinct.sortBy(_._2.desc).take(getFilterLimit(filter)).result)
+    val op = dbHandle.run(action.distinct.sortBy(_._2.desc).take(1000).result)
```

Comment:
Instead of hard coding the number 1,000, let's instead read it from the conf file.